storage/schemadb/tests/db.rs

```diff
@@ -384,3 +384,22 @@ fn test_report_size() {
         0
     );
 }
+
+#[test]
+fn test_checkpoint() {
+    let tmpdir = diem_temppath::TempPath::new();
+    let checkpoint = diem_temppath::TempPath::new();
+    {
+        let db = open_db(&tmpdir);
+        db.put::<TestSchema1>(&TestField(0), &TestField(0)).unwrap();
+        db.create_checkpoint(&checkpoint).unwrap();
+    }
+    {
+        let db = open_db(&checkpoint);
+        assert_eq!(
+            db.get::<TestSchema1>(&TestField(0)).unwrap(),
+            Some(TestField(0)),
+        );
+        db.put::<TestSchema1>(&TestField(1), &TestField(1)).unwrap();
+    }
+}
```

Comment:
Test that writing to the checkpoint didn't affect the original DB (by reading 0 and 1)?